overlord/snapstate/target.go

```diff
@@ -754,3 +818,467 @@ func validatedComponentInfo(path string, si *snap.Info, csi *snap.ComponentSideI
 	}
 	return componentInfo, nil
 }
+
+// UpdateSummary contains the data that describes an update, including a list of
+// Target structs that represent the snaps that are to be updated.
+type UpdateSummary struct {
+	// Requested is the list of snaps that were requested to be updated. If
+	// RefreshAll is true, then this list should be empty.
+	Requested []string
+	// RefreshAll is true if all snaps on the system are being refreshed (could
+	// be either an auto-refresh or something like a manual "snap refresh").
+	// This mostly has the effect of ignoring some some non-fatal errors.
+	RefreshAll bool
```

Comment:
can this be false with Requested being empty?

Reply:
No, they're exclusive. Could just be a method on `UpdateSummary` (or whatever we call it) so they're always tied together?

Reply:
thanks, yes, I think a method would make more sense for this